[ews-build.webkit.org] [WPE] Add build bot using Skia backend

dpino opened the pull request:
#### 00f4aff6812b504f7508f250a635a2a6ff93fd51
<pre>
[ews-build.webkit.org] [WPE] Add build bot using Skia backend
<a href="https://bugs.webkit.org/show_bug.cgi?id=268807">https://bugs.webkit.org/show_bug.cgi?id=268807</a>

Reviewed by Adrian Perez de Castro.

Add new EWS build bot for testing WPE with Skia backend.

* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps_unittest.py:
(TestCheckChangeRelevance.test_queues_without_relevance_info):

Canonical link: <a href="https://commits.webkit.org/274492@main">https://commits.webkit.org/274492@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/54a812c95659564692b397d8a73079b545ee22e9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38089 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17000 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/40358 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40636 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33876 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/40214 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19723 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/14340 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/32160 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38663 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14359 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/33336 "Build was cancelled. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Skipping applying patch since patch_id isn't provided; Checked out pull request") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12520 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12452 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/34067 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41912 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/34583 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34528 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38333 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/13059 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10728 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36526 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/38136 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14607 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8784 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13472 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/14071 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Tools/CISupport/ews-build/config.json:
```diff
@@ -391,6 +395,12 @@
+    {
+      "name": "WPE-Skia-Build-EWS", "shortname": "wpe-skia", "icon": "buildOnly",
+      "factory": "WPEBuildFactory", "platform": "wpe",
+      "configuration": "release", "architectures": ["x86_64"],
```
We will want to pass `USE_SKIA=ON` to the build step, which could be done the [same way as for the post-commit queue, using additionalArguments](https://github.com/WebKit/WebKit/pull/24146/files):

```suggestion
      "configuration": "release", "architectures": ["x86_64"],
      "additionalArguments": ["--cmakeargs=-DUSE_SKIA=ON"],
```

dpino replied:
It's not needed. Bots can set additional cmakeargs via BUILD_WEBKIT_ARGS env variable. I'm still fine with the change because since this a Skia bot setting the argument in the `config.json` file makes it more explicit.

dpino replied:
I changed my mind, I think it's not a good idea to put cmakeargs arguments in the `config.json` bot.

Currently there's no other EWS bot that does that. As for `Tools/CISupport/build-webkit-org/config.json` there's currently only two bots that do that (both WPE bots).

The disadvantge of putting build arguments in the `config.json` file is that an build argument change (add, remove or modify build arguments) implies a commit in `config.json`. I believe that's the reason why we tend not to put build arguments in `config.json`.

donny-dont replied:
I'd like to add a `--skia` argument to `build-webkit` so more on our side can start playing with it.

dpino replied:
The equivalent change for https://github.com/WebKit/WebKit/pull/24146 didn't land in the end so I assume the same change for ews is not needed.

aperezdc reviewed (changes requested):
For the WPE Skia queue we will want to pass `USE_SKIA=ON` to CMake.

dpino committed:
[ews-build.webkit.org] [WPE] Add build bot using Skia backend
https://bugs.webkit.org/show_bug.cgi?id=268807

Reviewed by Adrian Perez de Castro.

Add new EWS build bot for testing WPE with Skia backend.

* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps_unittest.py:
(TestCheckChangeRelevance.test_queues_without_relevance_info):

Canonical link: https://commits.webkit.org/274492@main
